conda_index/index/__init__.py

```diff
@@ -743,17 +731,54 @@ def extract_wrapper(args):
                         log.debug("%s write index.html", subdir)
                         self._write_subdir_index_html(subdir, patched_repodata)
 
-                        t2.set_description("Updating channeldata")
-                        t2.update()
-                        log.debug("%s update channeldata", subdir)
-                        self._update_channeldata(channel_data, patched_repodata, subdir)
-
                         log.debug("%s finish", subdir)
 
-                # Step 7. Create and write channeldata.
-                self._write_channeldata_index_html(channel_data)
-                log.debug("write channeldata")
-                self._write_channeldata(channel_data)
+    def update_channeldata(self):
+        """
+        Update channeldata based on re-reading output `repodata.json` and existing
+        `channeldata.json`. Call after index() if channeldata is needed.
+        """
+        subdirs = self.detect_subdirs()
+
+        # Step 1. Lock local channel.
```

Comment:
Nit: continuity.  This is commented as `Step 1` and the next step is `Step 7`.  :confused-dog:

Reply:
Probably delete these comments now that we've reduced steps